leftwm-core/src/models/tag.rs

```diff
@@ -242,6 +242,23 @@ impl Tag {
                 .for_each(|w| {
                     w.set_visible(true);
                 });
+        } else if let Some(window) = windows
+            .iter_mut()
+            .find(|w| w.has_tag(&self.id) && w.is_maximized())
+        {
+            window.set_visible(true);
+            window.normal = Xyhw::from(workspace.rect());
+            let handle = window.handle;
+            windows
+                .iter_mut()
+                .filter(|w| {
+                    w.has_tag(&self.id)
+                        && w.transient.unwrap_or_else(|| 0.into()) == handle
+                        && w.is_managed()
+                })
+                .for_each(|w| {
+                    w.set_visible(true);
+                });
```

Comment:
This code is mainly repeating the above one, so how about 
```suggestion
        } else if let Some(window) = windows
            .iter_mut()
            .find(|w| w.has_tag(&self.id) && (w.is_maximized() || w.is_fullscreen()))
        {
            window.set_visible(true);
            window.normal = if w.is_fullscreen() { 
                workspace.xywh 
            } else { 
                Xyhw::from(workspace.rect())
            };
            let handle = window.handle;
            windows
                .iter_mut()
                .filter(|w| {
                    w.has_tag(&self.id)
                        && w.transient.unwrap_or_else(|| 0.into()) == handle
                        && w.is_managed()
                })
                .for_each(|w| {
                    w.set_visible(true);
                });
```
Note that this only works because you can only have one window either maximized or in fullscreen.

Reply:
I still want to add different behaviour to maximized windows, for example the ability to cycle between windows without reordering them (still figuring it out), so I'm currently working on how that works, after I finish it, I will try to remove duplicated code and stuff